In [1]:
import json
import pandas as pd

# Explore type of videos

In [19]:
with open("../data/GB_category_id.json") as json_input:
    categories = json.load(json_input)

categories_mapping = {29:'Unkown'}
print("Categories:")
for c in categories['items']:
    print(f"{c['snippet']['title']}, id: {c['id']}")
    categories_mapping[int(c['id'])] = c['snippet']['title']
    
print(categories_mapping)
#Music, id: 10

Categories:
Film & Animation, id: 1
Autos & Vehicles, id: 2
Music, id: 10
Pets & Animals, id: 15
Sports, id: 17
Short Movies, id: 18
Travel & Events, id: 19
Gaming, id: 20
Videoblogging, id: 21
People & Blogs, id: 22
Comedy, id: 23
Entertainment, id: 24
News & Politics, id: 25
Howto & Style, id: 26
Education, id: 27
Science & Technology, id: 28
Movies, id: 30
Anime/Animation, id: 31
Action/Adventure, id: 32
Classics, id: 33
Comedy, id: 34
Documentary, id: 35
Drama, id: 36
Family, id: 37
Foreign, id: 38
Horror, id: 39
Sci-Fi/Fantasy, id: 40
Thriller, id: 41
Shorts, id: 42
Shows, id: 43
Trailers, id: 44
{29: 'Unkown', 1: 'Film & Animation', 2: 'Autos & Vehicles', 10: 'Music', 15: 'Pets & Animals', 17: 'Sports', 18: 'Short Movies', 19: 'Travel & Events', 20: 'Gaming', 21: 'Videoblogging', 22: 'People & Blogs', 23: 'Comedy', 24: 'Entertainment', 25: 'News & Politics', 26: 'Howto & Style', 27: 'Education', 28: 'Science & Technology', 30: 'Movies', 31: 'Anime/Animation', 32: 'Action/Adventur

In [42]:
column_list = ['video_id', 'title', 'channel_title', 'category_id']
videos = pd.read_csv("../data/GBvideos.csv", usecols=column_list)
print(videos.columns)

videos.describe()

Index(['video_id', 'title', 'channel_title', 'category_id'], dtype='object')


category_id
count  7993.000000
mean     19.738271
std       7.178132
min       1.000000
25%      17.000000
50%      23.000000
75%      24.000000
max      29.000000

Many entries are repeated for the same videos but taken in different dates. 
We will keep only one entry for each videoID

In [36]:
videos = videos.drop_duplicates()

In [40]:
#Add category column as a name
category_name = []
for index, row in videos.iterrows():
    category_name.append(categories_mapping[int(row['category_id'])])

videos['category_name'] = category_name

videos.describe()

category_id
count  1776.000000
mean     19.766329
std       7.025187
min       1.000000
25%      17.000000
50%      22.000000
75%      24.000000
max      29.000000

In [43]:
videos.head()

video_id                                              title  \
0  jt2OHQh0HoQ  Live Apple Event - Apple September Event 2017 ...   
1  AqokkXoa7uE  Holly and Phillip Meet Samantha the Sex Robot ...   
2  YPVcg45W0z4                    My DNA Test Results! I'm WHAT?!   
3  T_PuZBdT2iM  getting into a conversation in a language you ...   
4  NsjsmgmbCfc                               Baby Name Challenge!   

       channel_title  category_id  
0        Apple Event           28  
1       This Morning           24  
2       emmablackery           24  
3              ProZD            1  
4  Sprinkleofglitter           26

In [38]:
#Count videos per category
videos.groupby(['category_id', 'category_name']).size()

category_id  category_name       
1            Film & Animation         78
2            Autos & Vehicles         14
10           Music                   288
15           Pets & Animals           21
17           Sports                  171
19           Travel & Events          10
20           Gaming                   67
22           People & Blogs          250
23           Comedy                  124
24           Entertainment           327
25           News & Politics          69
26           Howto & Style           256
27           Education                46
28           Science & Technology     51
29           Unkown                    4
dtype: int64

288 are music videos

# Explore the number of comments for a given category